src/common/src/array/stream_chunk.rs

```diff
@@ -370,7 +370,7 @@ impl StreamChunk {
         } else {
             false
         };
-        if chunk_size == limit + 1 && ends_with_update {
+        if chunk_size == burst + 1 && ends_with_update {
```

Comment:
given that we will never split a chunk, we do not need to check whether the last message is Update?

Reply:
Yes this logic seems outdated for source rate limit, but it may be still required by other rate limit logic like backfill rate limit or something.

Reply:
After https://github.com/risingwavelabs/risingwave/pull/19967, rate limiters are replaced with our own implementation, which is able to handle unlimited burst. So this can indeed be removed.